test/unit/client.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+/*global expect, describe, it*/
+const Client = require("../../lib/client");
+const chai = require("chai");
+
+describe('client', function () {
+  it('should append error details when available', async () => {
+    const client = new Client({ client: {} })
+    client._query = () => {
+      const err = new Error('error message')
+      err.detail = 'hmm very tasty details'
+      return Promise.reject(err)
+    }
+    try {
+      await client.query('select * from a', {sql: 'select * from a', bindings: []})
+    } catch (err) {
```

Comment:
wouldn't this also pass if no error is thrown altogether? See https://stackoverflow.com/questions/26571328/how-do-i-properly-test-promises-with-mocha-and-chai for approaches to test throwing promises (see second answer).

Reply:
there is an error thrown in this test always here on line 12: https://github.com/tgriesser/knex/pull/2598/files/fe87405f1bbfd0b561627dceb8c5d63b560726c2#diff-cde1f14853c6f9159fe30b7e09ffdd92R12

If no error is thrown the line of code I added is not executed. Do you want me to write unit tests for code I did not commit? I can do it, but I don't really understand what it has do to with the change I opened.

Reply:
Fair enough, just pointing out how to decrease amount of assumptions one has to make when reading or maintaining these tests, but it's really a minor nitpick. LGTM
@elhigu Do you think this can be merged?